docs/dev/stream/state/state.md

```diff
@@ -266,6 +266,92 @@ a `ValueState`. Once the count reaches 2 it will emit the average and clear the
 we start over from `0`. Note that this would keep a different state value for each different input
 key if we had tuples with different values in the first field.
 
+### State time-to-live (TTL)
+
+A time-to-live (TTL) can be assigned to the keyed state value. 
+In this case it will expire after the configured TTL
+and its stored value will be cleaned up based on the best effort.
```

Comment:
I was expecting some additional info about what "best effort" means here, but it came a bit later. Maybe we should say that this is discussed in more detail later. I think it might also make sense to name some use cases that are currently supported (and some that are not/will be in the future) to make things a bit more concrete for users.